cfme/storage/object_store_object.py

```diff
@@ -111,10 +113,24 @@ class ObjectStoreObjectCollection(BaseCollection):
     def all(self):
         """returning all Object Store Objects"""
         view = navigate_to(self, 'All')
-        view.toolbar.view_selector.select("Grid View")
-        objects = [self.instantiate(key=item)
-                   for item in view.entities.entity_names]
-        return objects
+        view.entities.paginator.set_items_per_page(500)
+        objects = []
+
+        try:
+            if 'provider'in self.filters:
+                for item in view.entities.elements.read():
+                    if self.filters.get('provider').name in item['Cloud Provider']:
```

Comment:
**REQUIRED**: Since you ensured `provider` indeed is in the `self.filters`, therefore you can use `[]`, `.get` is not necessary.